src/docfx/build/page/PageBuilder.cs

```diff
@@ -261,6 +266,27 @@ private SystemMetadata CreateSystemMetadata(ErrorBuilder errors, FilePath file,
         return systemMetadata;
     }
 
+    private string? GetCanonicalUrlWithMonikerIfNecessary(string? pageType, FilePath file)
+    {
+        if (pageType == "profile")
+        {
+            return null;
+        }
+
+        var canonicalUrl = _documentProvider.GetCanonicalUrl(file);
+
+        if (_config.IsCanonicalUrlWithMoniker)
+        {
+            var canonicalVersion = _publishUrlMap.GetCanonicalVersion(file);
+            if (!string.IsNullOrEmpty(canonicalVersion))
+            {
+                return $"{canonicalUrl}?view={Uri.EscapeDataString(canonicalVersion)}";
```

Comment:
Should use the same logic as in `EncodePath`

Reply:
sure